src/jspm-hmr-server.ts

```diff
@@ -66,7 +66,7 @@ export function createServer(options: ServerOptions): JspmHmrServer {
 
     const proxy = httpProxy.createProxyServer();
     app.use(proxyRoute, (req, res) => {
-      req.url = req.baseUrl;
+      req.url = `${req.baseUrl}${req.url}`;
```

Comment:
@amygdaloideum added test coverage for your case here: https://github.com/piotrwitek/jspm-hmr/pull/18
confirmed to be working as expected :)